### ISSUE TITLE
Missing support for Spring Data REST

### ISSUE THREAD

**Issue by davioooh** [pac4j/spring-webmvc-pac4j#20]
I'm trying to implement JWT authentication in a Spring Data REST project, but I noticed that spring-webmvc-pac4j library is working only with _regular_ controllers and not for the endpoints generated by Spring Data REST framework.

Is there a way to enable this, or is this feature missing?

Thanks,
David

**Comment by leleuj** [comment]
To apply security in Spring MVC, pac4j has a `SecurityInterceptor` which is defined on URL patterns. So it should work for any controller or REST API.
Example:

```java
registry.addInterceptor(new SecurityInterceptor(config, "jwt")).addPathPatterns("/api/*");
```

**Comment by davioooh** [comment]
Thank you @leleuj , I have already configured a `SecurityInterceptor` in my project, but it's not working...

Some additional detail:

In my project I'm using [Spring Data REST](https://projects.spring.io/spring-data-rest/) to dynamically generate endpoints for the REST API.  All these endpoints match this pattern: `/api/*`

Then I manually created a test rest controller, like this:

```
@RestController
public class TestController {
    @GetMapping("/test")
    public Item testMethod(){
        return new Test();
    }
}
```

As you suggested I added a` SecurityInterceptor` in my configuration:

`registry.addInterceptor(new SecurityInterceptor(config, "HeaderClient")).addPathPatterns("/*");`

and ran my project.

As checked before. The `/test` endpoint is secured, while all the `/api/*` path are still accessible...

Is there something wrong in the way I am configuring path pattern?
